src/auxiliary/t8code.jl

```diff
@@ -95,29 +145,58 @@ function trixi_t8_count_interfaces(forest)
 
                 forest_is_balanced = Cint(1)
 
-                t8_forest_leaf_face_neighbors(forest, itree, element,
+                t8_forest_leaf_face_neighbors(mesh.forest, itree, element,
                                               pneighbor_leafs_ref, iface, dual_faces_ref,
                                               num_neighbors_ref,
                                               pelement_indices_ref, pneigh_scheme_ref,
                                               forest_is_balanced)
 
                 num_neighbors = num_neighbors_ref[]
+                dual_faces = unsafe_wrap(Array, dual_faces_ref[], num_neighbors)
                 neighbor_ielements = unsafe_wrap(Array, pelement_indices_ref[],
                                                  num_neighbors)
                 neighbor_leafs = unsafe_wrap(Array, pneighbor_leafs_ref[], num_neighbors)
                 neighbor_scheme = pneigh_scheme_ref[]
 
-                if num_neighbors > 0
+                if num_neighbors == 0
+                    local_num_boundary += 1
+                else
                     neighbor_level = t8_element_level(neighbor_scheme, neighbor_leafs[1])
 
-                    # Conforming interface: The second condition ensures we only visit the interface once.
-                    if level == neighbor_level && current_index <= neighbor_ielements[1]
-                        local_num_conform += 1
-                    elseif level < neighbor_level
-                        local_num_mortars += 1
+                    if all(neighbor_ielements .< num_local_elements)
+                        # Conforming interface: The second condition ensures we
+                        # only visit the interface once.
+                        if level == neighbor_level && current_index <= neighbor_ielements[1]
+                            local_num_conform += 1
+                        elseif level < neighbor_level
+                            local_num_mortars += 1
+                            # else `level > neighbor_level` is ignored.
```

Comment:
Maybe elaborate why?

Reply:
The mortar interface is only counted once which is already covered by the line above. I added a comment.